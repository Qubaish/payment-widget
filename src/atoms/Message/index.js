import React from 'react';
import styled from 'styled-components';


const StyledMsg = styled.div`
    background: ${props => props.isError ? '#F8D7D9' : '#D3E2F3'};
    height: 50px;
    line-height: 50px;
    color: ${props => props.isError ? '#731C23' : '#609EF4'};
    text-align: left;
    padding-left: 2rem;
    border-radius: 10px;
`;

function Message({textMsg, isError}) {
    return (
        <StyledMsg isError={isError}>{textMsg}</StyledMsg>
    );
}

Message.defaultProps = {
    textMsg: 'Valid OTP number!'
};

export default Message;