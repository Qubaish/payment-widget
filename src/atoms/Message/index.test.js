import React from 'react';
import { render } from '@testing-library/react';
import Message from './index';

describe('<Message />', () => {
    it('checking snapshot', () => {
        const { asFragment } = render(<Message textMsg="hello"/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render a message', () => {
        const { getByTestId} = render(<Message textMsg="test"/>);
        const elem = getByTestId('msg');
        expect(elem.innerHTML).toBe('test');
    });
    it('should error type message', () => {
        const { getByTestId} = render(<Message textMsg="test" isError={true}/>);
        const elem = getByTestId('msg');
        expect(elem).toHaveStyle('background: #F8D7D9');
    });
});