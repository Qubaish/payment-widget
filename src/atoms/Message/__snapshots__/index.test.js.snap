// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> checking snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm jHqMMU"
    data-testid="msg"
  >
    hello
  </div>
</DocumentFragment>
`;
